dev-libs/caf/caf-0.17.3.ebuild

```diff
@@ -0,0 +1,116 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+CMAKE_ECLASS=cmake
+PYTHON_COMPAT=( python3_{6,7,8} )
+inherit cmake-multilib python-single-r1
+
+DESCRIPTION="The C++ Actor Framework (CAF)"
+HOMEPAGE="https://actor-framework.org/"
+SRC_URI="https://github.com/actor-framework/actor-framework/archive/${PV}.tar.gz -> ${P}.tar.gz"
+LICENSE="|| ( Boost-1.0 BSD )"
+SLOT="0/17.3"
+KEYWORDS="~amd64 ~x86"
+IUSE="debug doc examples opencl +openssl python static-libs test tools"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+RDEPEND="examples? ( net-misc/curl[${MULTILIB_USEDEP}] )
+	openssl? ( dev-libs/openssl:0=[${MULTILIB_USEDEP},static-libs?] )
+	opencl? ( virtual/opencl[${MULTILIB_USEDEP}] )
+	python? ( ${PYTHON_DEPS}
+		dev-python/ipython
+		dev-python/pybind11[${PYTHON_USEDEP}] )"
+
+DEPEND="${RDEPEND}"
+
+BDEPEND="doc? ( app-doc/doxygen[dot]
+	app-shells/bash:0
+	dev-texlive/texlive-fontsextra
+	dev-texlive/texlive-latex
+	dev-texlive/texlive-latexextra )"
+
+RESTRICT="!test? ( test )"
+
+src_unpack() {
+	default
+	mv * "${P}" || die
+}
+
+src_prepare() {
+	append-cflags "-std=c++11 -Wextra -Wall -pedantic"
+	append-cxxflags "-std=c++11 -Wextra -Wall -pedantic"
```

Comment:
Why do we force these flags?

Reply:
provided build system does this:

```
if(NOT CMAKE_CXX_FLAGS)
  set(CMAKE_CXX_FLAGS                   "-std=c++11 -Wextra -Wall -pedantic ${EXTRA_FLAGS}")
```
but of course we provide our own cmake cxx flags, so that does not happen. So I thought it's wise to replicate. Happy to remove, please advice.

Reply:
Ok if that mimics upstream, thanks.

Reply:
I've a pretty default make.conf:
````
COMMON_FLAGS="-mtune=native -O2 -pipe"
CFLAGS="${COMMON_FLAGS}"
CXXFLAGS="${COMMON_FLAGS}"
````

With these settings it's transformed to:
````
...
-- Supplied CXXFLAGS do not contain a C++ standard, setting std to c++11
...
CXXFLAGS:              -std=c++11 -mtune=native -O2 -pipe 
...
C               -mtune=native -O2 -pipe
C++             -std=c++11 -mtune=native -O2 -pipe
````

Which seems to work fine on my testing machine and it respects my settings. If it's not causing issues, I'd remove these append-* calls.

Reply:
Sounds good, done.